### ISSUE TITLE
2.4.5 Dev Pool Switching the Algo

### ISSUE THREAD

**Issue by redscream** [fireice-uk/xmr-stak#1671]
Please provide as much as possible information to reproduce the issue.

# Basic information
  - Type of the CPU.
No CPU Mining
  - Type of the GPU (if you try to miner with the GPU).
AMD
# Compile issues
  - Which OS do you use?
  Windows
  add **all** commands you used and the **full** compile output here
  ```
  ```
  run `cmake -LA .` in the build folder and add the output here
  ```

# Issue with the execution
  - Do you compiled the miner by our own?
  NO
  run `./xmr-stak --version-long` and add the output here
  ```

# AMD OpenCl issue

  ```
  run `clinfo` and add the output here
  ```

# Stability issue
  - Is the CPU or GPU overclocked?
  - Is the Main memory of the CPU or GPU undervolted?

Mining the CN-Fast Algo, however Dev pool is CN-V7 which means xmr-stak switches the algo on the fly this causes stability issues... Hash drops, GPU failures ect.

**Comment by psychocrypt** [comment]
Yes I mentioned it in the pull request that the hashrate will be wrong if
the dev pool is active. But it will not have an effect on the user pool.
The hashrate is still what the miner can mine.
What kind of stability issues do you have?

**Comment by redscream** [comment]
Hi,
Thanks for your response. As you are aware as we are rx vega owners mine on
windows and the hash stability is not guaranteed. It is needed to restart
windows for fixing any issue. On the new version 2.4.5 it changes the
algorithm to V7 for dev fee so rx vegas loses their max hashes, and mining
low performance and get more rejects from user pool . Please kindly do not
change the algorithm on the fly for dev fee. I am ok to mine dev pool with
my current algorithm for my farm stability.
Kind regards...

2018-06-21 19:38 GMT+03:00 psychocrypt <notifications@github.com>:

**Comment by Schmoozer0069** [comment]
I have noticed a gradual drop in hash rates when the miner switches from the dev pool back to the miner pool.  Over the course of 6-8 hours my rig will drop 1500 - 2500 H/s.  I have not noticed any drop in hash rates when mining any other CN algo.  I have tried using precious guardian to fix this.

**Comment by redscream** [comment]
Hashes reduce after switch back to user pool. Since algorithm is changing
(user pool cn-fast, dev pool cn 7) If you define a dev pool with cn-fast
for who mines masari, the issue will be solved I believe.

12 Tem 2018 Per, saat 15:27 tarihinde psychocrypt <notifications@github.com>
şunu yazdı:

**Comment by Schmoozer0069** [comment]
As redscream noted, there is a noticeable drop in hash rate when switching back to CN fast.  After running XMR-Stak for 16 hours my hash rate went from 21 KH/s to 15 KH/s.  My other post has more accurate numbers, but you should get an idea as to the issue.  I am thinking it has more to do with the actual algo than the mining software.  I ran Cast for 16 hours and got very similar results, even using ratewatch.
